src/amino/arkeo/index.ts

```diff
@@ -0,0 +1,359 @@
+                creator,
+                provider,
+                service,
+                metadataUri,
+                metadataNonce,
+                status,
+                minContractDuration,
+                maxContractDuration,
+                subscriptionRate,
+                payAsYouGoRate,
+                settlementDuration,
+            }: codecs.arkeo.arkeo.MsgModProvider): AminoMsgModProvider['value'] => ({
+                creator: toBech32('arkeo', creator),
+                provider: toBech32('arkeo', provider),
+                service: service,
+                metadata_uri: metadataUri,
+                metadata_nonce: metadataNonce,
+                status: status,
+                min_contract_duration: minContractDuration,
+                max_contract_duration: maxContractDuration,
+                subscription_rate: subscriptionRate,
+                pay_as_you_go_rate: payAsYouGoRate,
+                settlement_duration: settlementDuration,
+            }),
+            fromAmino: ({
+                creator,
+                provider,
+                service,
+                metadata_uri,
+                metadata_nonce,
+                status,
+                min_contract_duration,
+                max_contract_duration,
+                subscription_rate,
+                pay_as_you_go_rate,
+                settlement_duration,
+            }: AminoMsgModProvider['value']): codecs.arkeo.arkeo.MsgModProvider => ({
+                creator: fromBech32(creator).data,
+                provider: fromBech32(provider).data,
+                service: service,
+                metadataUri: metadata_uri,
+                metadataNonce: metadata_nonce,
+                status: status,
+                minContractDuration: min_contract_duration,
+                maxContractDuration: max_contract_duration,
+                subscriptionRate: subscription_rate,
+                payAsYouGoRate: pay_as_you_go_rate,
+                settlementDuration: settlement_duration,
+            }),
+        },
+        '/arkeo.arkeo.MsgOpenContract': {
+            aminoType: 'arkeo/arkeo/MsgOpenContract',
+            toAmino: ({
+                creator,
+                provider,
+                service,
+                client,
+                delegate,
+                contractType,
+                duration,
+                rate,
+                deposit,
+                settlementDuration,
+                authorization
+            }: codecs.arkeo.arkeo.MsgOpenContract): AminoMsgOpenContract['value'] => ({
+                creator: toBech32('arkeo', creator),
+                provider: toBech32('arkeo', provider),
+                service: service,
+                client: toBech32('arkeo', client),
+                delegate: toBech32('arkeo', delegate),
+                contract_type: contractType,
+                duration: duration,
+                rate: rate ?? {amount: '0', denom: 'uarkeo'},
```

Comment:
this strikes me as odd. seems like there should be some validation outside of the amino converters rather than the converter having side effects on the data.